packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -97,6 +97,7 @@ function getSelectionList(tree: CatalogTreeType): CatalogSelection[] {
 }
 
 export const CatalogPanel: FC = () => {
+    const [, startTransition] = useTransition();
```

Comment:
could you explain again what's the benefit of using this ? I can't notice any change even on slow 3g network settings.

Reply:
Sure! 
This helps when opening/closing the sidebar. You can notice that when opening/closing the sidebar and testing out the before and after - specially on a 3g network. This is also visible in our own Analytics instance.

Since the value is deferred, the UI thread won't be blocked when processing the `metatada` - from the API and then the calculation in the `useMemo` in `CatalogMetadata`